examples/regression/1-quickstart/plot_compare_conformity_scores.py

```diff
@@ -0,0 +1,197 @@
+        y_pis_absconfscore[:, 1, 0].T - np.expand_dims(y_pred_absconfscore, 0),
+    ],
+    axis=0,
+)
+conf_int_width_absconfscore = (
+    y_pis_absconfscore[:, 1, 0] - y_pis_absconfscore[:, 0, 0]
+)
+coverage_absconfscore = regression_coverage_score(
+    y_test, y_pis_absconfscore[:, 0, 0], y_pis_absconfscore[:, 1, 0]
+)
+
+# Train model with GammaConformityScore
+mapie = MapieRegressor(
+    RandomForestRegressor(**rf_kwargs), conformity_score=GammaConformityScore()
+)
+mapie.fit(X_train, y_train)
+y_pred_gammaconfscore, y_pis_gammaconfscore = mapie.predict(
+    X_test, alpha=[alpha]
+)
+yerr_gammaconfscore = np.concatenate(
+    [
+        np.expand_dims(y_pred_gammaconfscore, 0)
+        - y_pis_gammaconfscore[:, 0, 0].T,
+        y_pis_gammaconfscore[:, 1, 0].T
+        - np.expand_dims(y_pred_gammaconfscore, 0),
+    ],
+    axis=0,
+)
+conf_int_width_gammaconfscore = (
+    y_pis_gammaconfscore[:, 1, 0] - y_pis_gammaconfscore[:, 0, 0]
+)
+coverage_gammaconfscore = regression_coverage_score(
+    y_test, y_pis_gammaconfscore[:, 0, 0], y_pis_gammaconfscore[:, 1, 0]
+)
+
+##############################################################################
+# 3. Compare the confidence intervals
+# --------------------------------------------------------------------------
+#
+# Once the models have been trained, we now compare the confidence intervals
+# obtained from the two conformity scores. We can see that the
+# AbsoluteConformityScore generates confidence interval with almost the same
+# width for all the predicted values. Converly, the GammaConformityScore
+# yields confidence interval with width scaling with the predicted values.
+#
+# The choice of the conformity score depends on the problem we face.
+
+fig, axs = plt.subplots(2, 2, figsize=(16, 16))
+
+axs[0, 0].errorbar(
+    y_test,
+    y_pred_absconfscore,
+    yerr=yerr_absconfscore,
+    alpha=0.5,
+    linestyle="None",
+)
+axs[0, 0].scatter(y_test, y_pred_absconfscore, s=1, color="black")
+axs[0, 0].plot(
+    [0, max(max(y_test), max(y_pred_absconfscore))],
+    [0, max(max(y_test), max(y_pred_absconfscore))],
+    "-r",
+)
+axs[0, 0].set_xlabel("actual price [$]")
+axs[0, 0].set_ylabel("pred price [$]")
+axs[0, 0].grid()
+axs[0, 0].set_title(
+    f"AbsoluteResidualScore - coverage={coverage_absconfscore:.0%}"
+)
+
+axs[0, 1].errorbar(
+    y_test,
+    y_pred_gammaconfscore,
+    yerr=yerr_gammaconfscore,
+    alpha=0.5,
+    linestyle="None",
+)
+axs[0, 1].scatter(y_test, y_pred_gammaconfscore, s=1, color="black")
+axs[0, 1].plot(
+    [0, max(max(y_test), max(y_pred_gammaconfscore))],
+    [0, max(max(y_test), max(y_pred_gammaconfscore))],
+    "-r",
+)
+axs[0, 1].set_xlabel("actual price [$]")
+axs[0, 1].set_ylabel("pred price [$]")
+axs[0, 1].grid()
+axs[0, 1].set_title(
+    f"GammaResidualScore - coverage={coverage_gammaconfscore:.0%}"
+)
+fig.suptitle(
+    f"Predicted values with the confidence intervals of level {alpha}"
+)
+
+xmin, xmax = axs[0, 0].get_xlim()
```

Comment:
Can the code be included in a loop or in a function for sake of compactness ?

Reply:
Done